File: app/src/main/java/com/dmitriykargashin/cardamontimecalculator/data/resultFormat/ResultFormat.kt
/*
 * Copyright (c) 2019. Dmitriy Kargashin
 */

package com.dmitriykargashin.cardamontimecalculator.data.resultFormat

import com.dmitriykargashin.cardamontimecalculator.data.tokens.Tokens

class ResultFormat(val formatTokens: Tokens, var convertedResultTokens: Tokens) {
    var isSelected: Boolean = false
    var textPresentationOfTokens: String = ""

    init {
        textPresentationOfTokens = formatTokens.toStringWithSpaces()
    }

    constructor (
        formatTokens: Tokens,
        convertedResultTokens: Tokens,
        exactlyTextPresentationOfTokens: String
    ) : this(formatTokens, convertedResultTokens) {
        textPresentationOfTokens = exactlyTextPresentationOfTokens
    }


}

File: flutter_app/lib/data/result_format.dart
import '../engine/tokens.dart';

/// One selectable output format. Port of Kotlin `ResultFormat`
/// (data/resultFormat/ResultFormat.kt).
class ResultFormat {
  /// [exactlyTextPresentationOfTokens] overrides the display label, which
  /// otherwise defaults to `formatTokens.toStringWithSpaces()` (used only by
  /// the "All Units" format).
  ResultFormat(
    this.formatTokens,
    this.convertedResultTokens, [
    String? exactlyTextPresentationOfTokens,
  ]) : textPresentationOfTokens =
            exactlyTextPresentationOfTokens ?? formatTokens.toStringWithSpaces();

  /// The ordered unit tokens of the format (e.g. "Hour Minute").
  final Tokens formatTokens;

  /// The current result rendered in this format (preview row content);
  /// replaced by ResultFormatsRepository.updateFormatsWithPreview.
  Tokens convertedResultTokens;

  /// Single-selection flag managed by ResultFormats.setSelection.
  bool isSelected = false;

  /// Display label (e.g. "Hour Minute", "All Units").
  String textPresentationOfTokens;
}
